File: testing/unit-test-patterns/c#/UnitTestPatterns/Common/Services/UniqueCloneValidationRulesRepository.cs
using System.Collections.Generic;
using UnitTestPatterns.Common.Interfaces;
using UnitTestPatterns.Common.Models;

namespace UnitTestPatterns.Common.Services
{
    public class UniqueCloneValidationRulesRepository : IUniqueCloneValidationRulesRepository
    {
        public IEnumerable<string> GetValidFirstNames(Gender gender)
        {
            // Make call to database and get list
            return new List<string>();
        }

        public IEnumerable<string> GetValidLastNames()
        {
            // Make call to database and get list
            return new List<string>();
        }

        public int GetMaximumNumberOfEyes()
        {
            // Make call to database and get value
            return 0;
        }
    }
}

File: testing/unit-test-patterns/javascript/src/common/services/UniqueCloneValidationRulesRepository.ts
import { IUniqueCloneValidationRulesRepository } from '../interfaces/UniqueCloneValidationRulesRepository.interface'
import { Gender } from '../models/gender'

export class UniqueCloneValidationRulesRepository
  implements IUniqueCloneValidationRulesRepository
{
  getValidFirstNames(gender: Gender): string[] {
    // Make call to database and get list
    return []
  }
  getValidLastNames(): string[] {
    // Make call to database and get list
    return []
  }
  getMaximumNumberOfEyes(): Promise<number> {
    // Make call to database and get value
    return Promise.resolve(0)
  }
}
